pkg/util/wait/wait_test.go

```diff
@@ -123,10 +123,11 @@ func TestPollImmediate(t *testing.T) {
 		invocations++
 		return true, nil
 	})
-	fp := fakePoller{max: 0, wg: sync.WaitGroup{}}
+	fp := fakePoller{max: 0}
 	if err := pollImmediateInternal(fp.GetWaitFunc(), f); err != nil {
 		t.Fatalf("unexpected error %v", err)
 	}
+	// We don't need to wait for fp.wg, as pollImmediate shouldn't call WaitFunc at all.
```

Comment:
It depends on what kind of error do we want to get if it'll get called - panic, or test failure. I can pass nil there if you prefer.